std/algorithm/comparison.d

```diff
@@ -780,7 +795,8 @@ template equal(alias pred = "a == b")
                 if (r2.empty) return false;
                 if (!binaryFun!(pred)(r1.front, r2.front)) return false;
             }
-            return r2.empty;
+            static if (!isInfinite!Range1)
+                return r2.empty;
```

Comment:
This is missing a return statement when `Range1` is infinite. Please add the return statement for that case and make sure that there are tests for both the case where lhs is infinite and rhs is finite (with no length) and the case where rhs is infinite and lhs is finite (with no length). The fact that there are no compilation errors for the tests indicate that at least the first of those two cases was missed.